util/matchers.go

```diff
@@ -0,0 +1,17 @@
+package util
+
+import (
+	"github.com/prometheus/prometheus/storage/metric"
+)
+
+// SplitFiltersAndMatchers splits empty matchers off, which are treated as filters, see #220
+func SplitFiltersAndMatchers(allMatchers []*metric.LabelMatcher) (filters, matchers []*metric.LabelMatcher) {
+	for _, matcher := range allMatchers {
+		if matcher.Match("") {
```

Comment:
Yep, it relies on the matcher being initialized via the constructor method (which should be the case), but otherwise it does the same.